src/overlays/actors/ovl_En_Slime/z_en_slime.c

```diff
@@ -88,103 +128,1169 @@ static DamageTable D_80A31B1C = {
+        random = Rand_ZeroOne();
+        effectVelocity.z = -Math_CosS(this->actor.world.rot.y) * ((random * 3.5f) + 1.0f);
+        effectVelocity.y = (Rand_ZeroOne() * 6.0f) + 2.0f;
+        EffectSsDtBubble_SpawnCustomColor(play, &effectPos, &effectVelocity, &sBubbleAccel, &sBubblePrimColor,
+                                          &sBubbleEnvColor, Rand_S16Offset(40, 20), 20, 0);
+    }
+
+    if (this->actor.colChkInfo.health == 0) {
+        Actor_PlaySfx(&this->actor, NA_SE_EN_SLIME_DEAD);
+    } else {
+        Actor_PlaySfx(&this->actor, NA_SE_EN_SLIME_DAMAGE);
+    }
+    this->actionFunc = EnSlime_UpdateDamageReaction;
+}
+
+/**
+ * Update reaction to taking damage.
+ * Updates the rebuff wobble. Once timer drops to 0, actor initializes death cycle,
+ * re-engages player, or returns to idle.
+ */
+void EnSlime_UpdateDamageReaction(EnSlime* this, PlayState* play) {
+    this->timer--;
+    Math_StepToF(&this->actor.speed, 0.0f, 1.0f);
+    if ((this->timer % 5) == 0) {
+        this->wobbleRot.x = Rand_ZeroOne() * (M_PI * 2.0f);
+        this->wobbleRot.z = Rand_ZeroOne() * (M_PI * 2.0f);
+    }
+    if (this->timer == 0) {
+        if (this->actor.colChkInfo.health != 0) {
+            this->collider.base.acFlags |= AC_ON;
+            if ((this->actor.xzDistToPlayer < 280.0f) && (Player_GetMask(play) != PLAYER_MASK_STONE)) {
+                EnSlime_EngagePlayer(this);
+            } else {
+                EnSlime_ReturnToIdle(this);
+            }
+        } else {
+            SoundSource_PlaySfxAtFixedWorldPos(play, &this->actor.world.pos, 40, NA_SE_EN_SLIME_BREAK);
+            EnSlime_InitializeDeath(this);
+        }
+    }
+}
+
+/**
+ * Initialize death action cycle. Disable AC_ON flag,
+ * make actor stop moving, and shrink the actor.
+ * Set action function to Expire.
+ */
+void EnSlime_InitializeDeath(EnSlime* this) {
+    this->collider.base.acFlags &= ~AC_ON;
+    if (this->actor.velocity.y > 0.0f) {
+        this->actor.velocity.y = 0.0f;
+    }
+    Actor_SetScale(&this->actor, 0.01f);
+    this->actionFunc = EnSlime_Expire;
+    this->actor.speed = 0.0f;
+}
+
+/**
+ * Spawn random effects (bubbles, splash) for death sequence.
+ * If chuchu is red, green, or yellow, spawn item drop.
+ * Shrink actor scale to 0.
+ * Call HideDead function for cleanup.
+ */
+void EnSlime_Expire(EnSlime* this, PlayState* play) {
+    s32 i;
+    Vec3f velocity;
+    Vec3f effectPos;
+    f32 random;
+    s32 factor;
+
+    if (Math_StepToF(&this->actor.scale.x, 0.0f, 0.002f)) {
+        effectPos.x = this->actor.world.pos.x;
+        effectPos.y = this->actor.world.pos.y + 20.0f;
+        effectPos.z = this->actor.world.pos.z;
+        EffectSsGSplash_Spawn(play, &effectPos, NULL, NULL, 1, 800);
+
+        for (i = 0; i < 15; i++) {
+            random = Rand_ZeroOne();
+
+            if (Rand_ZeroOne() < 0.5f) {
+                factor = -1;
+            } else {
+                factor = 1;
+            }
```

Comment:
This isn't really a factor, it's more of a sign. Both EnBigslime and EnMinislime have code similar to this, but they call it `randFloat` and `randSign`. From EnMinislime:
```c
                randFloat = Rand_ZeroFloat(invFrozenTimer);
                randSign = Rand_ZeroOne() < 0.5f ? -1 : 1;
```
What do you think?